plugins/org.python.pydev/tests/org/python/pydev/editor/correctionassist/heuristics/AssistFStringTest.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2020 by Brainwy Software Ltda
+ * Licensed under the terms of the Eclipse Public License (EPL).
+ * Please see the license.txt included with this distribution for details.
+ * Any modifications to this file must keep this entire header intact.
+ */
+package org.python.pydev.editor.correctionassist.heuristics;
+
+import java.util.List;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.Document;
+import org.python.pydev.core.MisconfigurationException;
+import org.python.pydev.core.docutils.PySelection;
+import org.python.pydev.core.proposals.CompletionProposalFactory;
+import org.python.pydev.editor.actions.PyAction;
+import org.python.pydev.editor.codecompletion.proposals.DefaultCompletionProposalFactory;
+import org.python.pydev.shared_core.code_completion.ICompletionProposalHandle;
+
+import junit.framework.TestCase;
+
+public class AssistFStringTest extends TestCase {
+    private AssistFString assist;
+
+    public static void main(String[] args) {
+        try {
+            AssistFStringTest test = new AssistFStringTest();
+            test.setUp();
+            test.tearDown();
+            junit.textui.TestRunner.run(AssistFStringTest.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /*
+     * @see TestCase#setUp()
+     */
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        assist = new AssistFString();
+        CompletionProposalFactory.set(new DefaultCompletionProposalFactory());
+    }
+
+    /*
+     * @see TestCase#tearDown()
+     */
+    @Override
+    protected void tearDown() throws Exception {
+        super.tearDown();
+        CompletionProposalFactory.set(null);
+    }
+
+    public void testSimple() throws BadLocationException, MisconfigurationException {
+        String d = "print('''a = %s %s''' % (a, b,))\n" +
```

Comment:
Nice... now that the basics are in place, let's extend it a bit more with a few cases:

Other cases to check: 

`"a = %s" % some_name`
`"a = %s" % some_name.another_name("ignore,anything,here")`
`"a = %s" % some_name[:2]`
`"a = %s %s"     %     (a  ,   c)`
`"a = %s %s" % some_name` // note: this doesn't need to be supported, just make sure that it doesn't give any error nor suggestion (whatever you think is easier -- if this was to be supported, the result would need to be:  `f"a = {some_name[0]} {some_name[1]}"`).